### DISCUSSION TITLE
In an eslint config, do 'extends' in an 'overrides' replace or do they merge with 'extends' up in the main section?

### DISCUSSION THREAD

**Discussion by miller-productions** [eslint/eslint#17174] [Q&A] [answered]
Hi folks,

I was wondering if someone would please be able to help me understand how multiple `extends` works in the context of `overrides`.

I'm wondering this: In an eslint config, say you have a value(s) in the main root `extends` section and also a value(s) down in an `extends` section within an `overrides`. Do the value(s) in `extends` in the `overrides` replace or do they merge with `extends` up in the main section?

To try and work it out myself, I had a go at playing around with various combinations of values in `extends` both in the main section of an eslint config and also down in the `overrides` section. However, I wasn't certain of the result/behaviour so I thought I'd ask.

I've seen that chiefmikey mentioned something in relation to his interpretation in [this post](https://github.com/eslint/eslint/discussions/15704#discussioncomment-2381420) where he said

> From what I understand now using an override just appends the extends array to the root config extends

What do you think? How does it work?

**Comment by aladdin-add** [comment, marked as answer]
> Do the value(s) in extends in the overrides replace or do they merge with extends up in the main section?

yes, it's merge, not replace.

**Reply by miller-productions** [reply]
Thanks, @aladdin-add. Much appreciated.
